torch/nn/attention/_templated_attention.py

```diff
@@ -79,17 +83,29 @@ def score_mod(
         Read more about feature classification at: https://pytorch.org/blog/pytorch-feature-classification-changes/#prototype
 
     """
+
+    if torch.compiler.is_dynamo_compiling():
+        out, _ = templated_attention_hop(query, key, value, score_mod)
+        return out
+
     # Some basic input validation
     _validate_sdpa_input(query, key, value)
     # This will restriction will be removed in newer version of the kernel
     if query.size(-2) != key.size(-2):
         raise ValueError(
             "NYI: The target sequence length (L) of the query tensor must match the source sequence length (S) of the key tensor."
         )
-    out, _ = templated_attention_hop(query, key, value, score_mod)
 
-    # Drop the logsumexp value since this is only needed for backwards
-    return out
+    if not torch._dynamo.is_dynamo_supported():
+        raise RuntimeError("templated attention requires dynamo support.")
+
+    with _set_compilation_env():
```

Comment:
REVIEW NOTE: We are requiring dynamo to be supported, this is needed to ensure that any closed over tensors are lifted correclty